src/main/java/com/thevoxelbox/voxelsniper/brush/type/SpiralStaircaseBrush.java

```diff
@@ -72,33 +72,47 @@ public List<String> handleCompletions(String[] parameters, Snipe snipe) {
         if (parameters.length == 1) {
             String parameter = parameters[0];
             return super.sortCompletions(Stream.of(
-                    "block", "step", "woodstair", "cobblestair",
-                    "c", "cc",
-                    "n", "e", "s", "w"
-            ), parameter, 0);
+                            "c", "cc",
+                            "n", "e", "s", "w"
+                    ), parameter, 0
+            );
         }
         return super.handleCompletions(parameters, snipe);
     }
 
     @Override
     public void handleArrowAction(Snipe snipe) {
         BlockVector3 targetBlock = getTargetBlock();
-        digStairWell(snipe, targetBlock); // make stairwell below target
+        digStairWell(snipe, targetBlock, getStairType(snipe)); // make stairwell below target
     }
 
     @Override
     public void handleGunpowderAction(Snipe snipe) {
         BlockVector3 lastBlock = getLastBlock();
-        buildStairWell(snipe, lastBlock); // make stairwell above target
+        buildStairWell(snipe, lastBlock, getStairType(snipe)); // make stairwell above target
+    }
+
+    /**
+     * @since 2.4.1
+     */
+    private StairType getStairType(Snipe snipe) {
+        BlockType blockType = snipe.getToolkitProperties().getBlockType();
+        if (Materials.isSlab(blockType)) {
+            return StairType.STEP;
+        }
+        if (Materials.isStair(blockType)) {
+            return StairType.STAIR;
+        }
+        return StairType.BLOCK;
     }
 
-    private void buildStairWell(Snipe snipe, BlockVector3 targetBlock) {
+    private void buildStairWell(Snipe snipe, BlockVector3 targetBlock, StairType stairType) {
         ToolkitProperties toolkitProperties = snipe.getToolkitProperties();
```

Comment:
It would probably make sense to refactor this into a private method, it shares the same ~50 lines like #digStairWell

Reply:
Makes sense ;)